libraries/AC_AutoTune/AC_AutoTune.cpp

```diff
@@ -1063,65 +866,66 @@ void AC_AutoTune::load_intra_test_gains()
     attitude_control->bf_feedforward(true);
     if (roll_enabled()) {
         attitude_control->get_rate_roll_pid().kP(orig_roll_rp);
-        attitude_control->get_rate_roll_pid().kI(orig_roll_rp*AUTOTUNE_PI_RATIO_FOR_TESTING);
+        attitude_control->get_rate_roll_pid().kI(get_intra_test_ri(axis));
         attitude_control->get_rate_roll_pid().kD(orig_roll_rd);
         attitude_control->get_rate_roll_pid().ff(orig_roll_rff);
         attitude_control->get_rate_roll_pid().filt_T_hz(orig_roll_fltt);
         attitude_control->get_rate_roll_pid().slew_limit(orig_roll_smax);
         attitude_control->get_angle_roll_p().kP(orig_roll_sp);
+        attitude_control->set_accel_roll_max_cdss(orig_roll_accel);
     }
     if (pitch_enabled()) {
         attitude_control->get_rate_pitch_pid().kP(orig_pitch_rp);
-        attitude_control->get_rate_pitch_pid().kI(orig_pitch_rp*AUTOTUNE_PI_RATIO_FOR_TESTING);
+        attitude_control->get_rate_pitch_pid().kI(get_intra_test_ri(axis));
         attitude_control->get_rate_pitch_pid().kD(orig_pitch_rd);
         attitude_control->get_rate_pitch_pid().ff(orig_pitch_rff);
         attitude_control->get_rate_pitch_pid().filt_T_hz(orig_pitch_fltt);
         attitude_control->get_rate_pitch_pid().slew_limit(orig_pitch_smax);
         attitude_control->get_angle_pitch_p().kP(orig_pitch_sp);
+        attitude_control->set_accel_pitch_max_cdss(orig_pitch_accel);
     }
     if (yaw_enabled()) {
         attitude_control->get_rate_yaw_pid().kP(orig_yaw_rp);
-        attitude_control->get_rate_yaw_pid().kI(orig_yaw_rp*AUTOTUNE_PI_RATIO_FOR_TESTING);
+        attitude_control->get_rate_yaw_pid().kI(get_intra_test_ri(axis));
         attitude_control->get_rate_yaw_pid().kD(orig_yaw_rd);
         attitude_control->get_rate_yaw_pid().ff(orig_yaw_rff);
         attitude_control->get_rate_yaw_pid().filt_T_hz(orig_yaw_fltt);
         attitude_control->get_rate_yaw_pid().slew_limit(orig_yaw_smax);
         attitude_control->get_rate_yaw_pid().filt_E_hz(orig_yaw_rLPF);
         attitude_control->get_angle_yaw_p().kP(orig_yaw_sp);
+        attitude_control->set_accel_yaw_max_cdss(orig_yaw_accel);
     }
 }
 
-// load_twitch_gains - load the to-be-tested gains for a single axis
+
+// load_test_gains - load the to-be-tested gains for a single axis
 // called by control_attitude() just before it beings testing a gain (i.e. just before it twitches)
-void AC_AutoTune::load_twitch_gains()
+void AC_AutoTune::load_test_gains()
```

Comment:
I think it is perhaps too confusing to split-up load_test_gains into three methods.  So we have the base method (this one), then overrides in the _Multi and _Heli classes which call the base.  I think it's just a bit too confusing for the reader to have to move back and forth between them to figure out what's going on.  Better to just force the sub classes to implement the method completely by adding a "=0" at the end of the definition in AC_AutoTune.h.  E.g. like this:

```
    // load gains for next test.  relies on axis variable being set
    virtual void load_test_gains();
```

Reply:
Done